extension/injections/extensible_injector.cpp

```diff
@@ -0,0 +1,131 @@
+//
+// Copyright (c) 2012-2017 Kanstantsin Chernik
+//
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+//<-
+#include <cassert>
+#include <memory>
+//->
+#include <boost/di.hpp>
+
+namespace di = boost::di;
+
+template <class TDep>
+class dependency_proxy
+    : di::core::dependency_base,
+      public di::core::dependency_impl<di::core::dependency_concept<typename TDep::expected, typename TDep::name>,
+                                       dependency_proxy<TDep>> {
+ public:
+  using scope = typename TDep::scope;
+  using expected = typename TDep::expected;
+  using given = typename TDep::given;
+  using name = typename TDep::name;
+  using priority = typename TDep::priority;
+  using scope_t = typename scope::template scope<expected, given>;
+
+  dependency_proxy(TDep& orig_dep) noexcept : ref_dep_(orig_dep) {}
+
+  dependency_proxy(dependency_proxy& other) noexcept : ref_dep_(other.ref_dep_) {}
+  dependency_proxy(dependency_proxy&& other) noexcept : ref_dep_(other.ref_dep_) {}
```

Comment:
Did you consider moving ref_dep_ in the move ctor?

Reply:
I can do it but what will happen to the reference I moved from?